fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/observer/eventmappings/AllNodeEventsOneEvent.java

```diff
@@ -78,21 +77,8 @@
                 final FedoraEvent fedoraEvent = from(evts.get(0));
                 evts.stream().forEach(evt -> {
                     // add types to the FedoraEvent from the JCR Event
-                    fedoraEvent.addType(FedoraEventImpl.valueOf(evt.getType()));
-                    try {
-                        // add appropriate properties to the FedoraEvent from the JCR Event
-                        if (evt.getPath().contains(JCR_CONTENT)) {
-                            fedoraEvent.addProperty("fedora:hasContent");
-                        }
-                        if (PROPERTY_EVENT_TYPES.contains(evt.getType())) {
-                            final String eventPath = evt.getPath();
-                            fedoraEvent.addProperty(eventPath.substring(eventPath.lastIndexOf('/') + 1));
-                        } else {
-                            LOGGER.trace("Not adding non-event property: {}, {}", fedoraEvent, evt);
-                        }
-                    } catch (final RepositoryException ex) {
-                        throw new RepositoryRuntimeException(ex);
-                    }
+                    fedoraEvent.getTypes().add(FedoraEventImpl.valueOf(evt.getType()));
+                    fedoraEvent.getResourceTypes().addAll(getResourceTypes(evt).collect(toSet()));
```

Comment:
No, actually those two lines are very important. The issue is this: say you have a lot of JCR events for a given node. These are collected into a single FedoraEvent in this class. But it turns out that those JCR events may have very different eventTypes and resourceTypes. These lines collect all of those into this FedoraEvent. You can try taking those lines out and seeing all the tests start to fail.